src/main/java/uk/gov/pay/ledger/transaction/service/TransactionMetadataService.java

```diff
@@ -62,7 +64,21 @@ public void upsertMetadataFor(Event event) {
         }
     }
 
-    public List<String> findMetadataKeysForTransactions(TransactionSearchParams searchParams) {
+    public void reprojectFromEventDigest(EventDigest eventDigest) {
+        var eventData = eventDigest.getEventPayload();
+
+        transactionDao.findTransactionByExternalId(eventDigest.getResourceExternalId())
+                .ifPresent(transactionEntity -> {
+                    @SuppressWarnings("unchecked")
+                    var metadata = (Map<String, Object>) eventData.get("external_metadata");
+                    metadata.forEach((key, value) -> {
```

Comment:
Can we check for null as external_metadata won't be available all the time and add relevant test ?

Reply:
🤦 Good spot! Updated